apps/api/src/report/types/milestone-table-entry.ts

```diff
@@ -0,0 +1,120 @@
+import { STATUS } from '@ien/common';
+
+/**
+ * milestone table entry for report 9 and 10
+ */
+export class MilestoneTableEntry {
+  id!: string;
+
+  registration_date!: Date;
+
+  [STATUS.APPLIED_TO_NNAS]: Date | null;
+
+  [STATUS.SUBMITTED_DOCUMENTS]: Date | null;
```

Comment:
actually, could we just use inheritance here and create a generic base class for all these statuses? `extends StatusClass<number> or StatusClass<Date | null>` ? or would that be too confusing

Reply:
I mean, these properties `STATUS.MILESTONE` are repeated in `MilestoneTableEntry` and `DurationTableEntry`.  They're pretty long, so my thought was to create a separate class for them, for ex.

`export class StatusListOrSomeName<T> {
  [STATUS.APPLIED_TO_NNAS]: T,
  [STATUS.SUBMITTED_DOCUMENTS]: T
........
..... the rest
}`

then we wouldn't have the same 30 or so lines in different files?

call like

`export class MilestoneTableEntry extends StatusListOrSomeName<Date> {` and `export class DurationTableEntry extends StatusListOrSomeName<number> {`

just a thought to reduce some lines

Reply:
Now I got it. Excellent! I'll try.

Reply:
Done